File: phalcon/util/Hmac.php
<?php
namespace Util;

/* 加密 */
class Hmac {

  /* Md5 */
  static function Md5(string $data): string {
    return md5($data);
  }

  /* Sha256 */
  static function Sha256(string $data, string $key): string {
    $res = hash_hmac('sha256', $data, $key, true);
    return $res!==false?$res:'';
  }

}

File: gin/util/hmac.go
package util

import (
	"crypto/hmac"
	"crypto/md5"
	"crypto/sha256"
	"encoding/hex"
)

/* Md5 */
func Md5(data string) string {
	h := md5.New()
	h.Write([]byte(data))
	return hex.EncodeToString(h.Sum(nil))
}

/* Sha256 */
func Sha256(data string, key string) string {
	h := hmac.New(sha256.New, []byte(key))
	h.Write([]byte(data))
	return hex.EncodeToString(h.Sum(nil))
}
